mycroft/client/speech/mic.py

```diff
@@ -797,6 +795,26 @@ def _wait_until_wake_word(self, source, sec_per_buffer):
                         # reset bytearray to store wake word audio in, else many
                         # serial detections
                         audio_buffer.clear()
+                    else:
+                        energy = SilenceDetector.get_debiased_energy(chunk)
+                        energies.append(energy)
+
+                        if len(energies) >= 4:
+                            # Adjust energy threshold once per second
+                            # avg_energy = sum(energies) / len(energies)
+                            max_energy = max(energies)
+                            self.silence_detector.current_energy_threshold = max_energy * 2
+
+                        if len(energies) >= 12:
```

Comment:
this is a port of mk2 code, which itself uses a hardcoded chunksize, but was tested in regular mycroft with a lower size and it worked fine. i dont think we need to do any changes here at least yet

lots of components depend on hardcoded sample rate/chunk size, this is non trivial to change since pockesphinx/precise/vad go bonkers

Reply:
Fair enough; I guess we can leave it to avoid diverging from Mycroft. My main worry is if we swap in some other WW plugin with different chunk size/SR expectations later, though I'm sure this isn't the only place we'd have to account for that